apiary.apib

```diff
@@ -542,7 +542,186 @@ API endpoints that Operation Code's Rails backend makes available to its React f
             errors: "Some error message"
         }
 
+## Mentor Request | Create a New Mentor Request [/api/v1/requests{?user_id,details,language,service_id,requested_mentor_id,status,assigned_mentor_id}]
++ Parameters
+    
+    + user_id (integer, required) - Identification of the User requesting a mentor in the form of an integer
+    + details (string, optional) - Details of the mentor request in the form of a string
+    + language (string, optional) - Primary language of the requesting mentee in the form of a string
+    + service_id (integer, optional) - Indentification of the mentorship request in the form of an integer
+    + requested_mentor_id (integer, optional) - Requested mentor's id in the form of an integer
+    + status (string, optional) - Current fullfillment status in the form on a string
+    + assigned_mentor_id (integer, optional) - Assigned mentor'd id in the form of an integer
+    
+    
+### Create a New Mentor Request [POST]
++ Request (application/json)
+
+    + Headers
+    
+            Authorization: Bearer Access-Token
+
+    + Body
+
+            {
+                "user_id": 4600,
+                "details": "I want to type well",
+                "language": "English",
+                "service_id": 200,
+                "requested_mentor_id": 250,
+                "status": "Open",
+                "assigned_mentor_id": 250
+            }
+
++ Response 201 (application/json)
+
+    + Body
+
+            {   
+                "id": 2130,
+                "user_id": 4600,
+                "details": "I want to type well",
+                "language": "English",
+                "service_id": 200,
+                "requested_mentor_id": 250,
+                "status": "Open",
+                "assigned_mentor_id": 250
+            }
+
++ Response 422 (application/json)
+
+    + Body
+
+            {
+                errors: "Some error message"
+            }
+  
+## Mentor Request | Show [/api/v1/requests/{request_id}]
++ Parameters
+    
+    + request_id (integer, required) - Indentification of mentor request in the form of an integer
+
+    
+### Show an Exiting Request From User ID [POST]
+
++ Request (application/json)
+
+    + Headers
+
+            Authorization: Bearer Access-Token
+
+    + Body
```

Comment:
In this case, the `id` is supplied as part of the `url` itself, and there will be no body in the request.

Reply:
I'm confused, so there's no body in the update of the request?

[Frontend](https://github.com/OperationCode/operationcode_frontend/blob/e2de0bc04de963aa8181ea2d7bae60f42460597e/src/shared/utils/apiHelper.js#L56), to me this looks like it's passing the id in the url like you say, but there's a body in the request that contains `status` and `mentor`. But then [here](https://github.com/wimo7083/operationcode_backend/blob/01e2a2d4c8fc91a869545ede3e32a1114bced5c5/app/controllers/api/v1/requests_controller.rb#L33) shows that the update takes in request params. 

I compared the syntax for both update with [Request](https://github.com/wimo7083/operationcode_backend/blob/master/app/controllers/api/v1/requests_controller.rb) controller and [User](https://github.com/wimo7083/operationcode_backend/blob/master/app/controllers/api/v1/users_controller.rb) controller and they both have the same type of input with params. The only difference is one uses `obj.update` and the other uses `obj.update_attributes`.

So i thought maybe this has to do with the difference. But [this](https://stackoverflow.com/a/27684300) states that the only difference in behavior of the two functions it the return value for a success vs failure. `update_attributes` returns false if it fails validation, `update` returns the new object if it fails validation. 

At this point I'm thorougly confused why there isn't a body in the request. 

I also used a `PUT` here but changed it to `PATCH` based on the information provided in the frontend axios call.

Reply:
@wimo7083 - Apologies, I miscommunicated that.  My intent was to say that the `id` would not be in the body, rather in the url itself.  And yes, you're correct, there is a body in the request with all the attributes you have below.